pkg/virt-controller/ipamclaims/ipamclaims.go

```diff
@@ -0,0 +1,185 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 Red Hat, Inc.
+ *
+ */
+
+package ipamclaims
+
+import (
+	"context"
+	"fmt"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	virtv1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/network/namescheme"
+	"kubevirt.io/kubevirt/pkg/network/vmispec"
+	"kubevirt.io/kubevirt/pkg/virt-controller/ipamclaims/libipam"
+	"kubevirt.io/kubevirt/pkg/virt-controller/network"
+
+	networkv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+
+	ipamclaims "github.com/k8snetworkplumbingwg/ipamclaims/pkg/crd/ipamclaims/v1alpha1"
+	ipamclaimsclient "github.com/k8snetworkplumbingwg/ipamclaims/pkg/crd/ipamclaims/v1alpha1/apis/clientset/versioned"
+
+	networkclient "kubevirt.io/client-go/generated/network-attachment-definition-client/clientset/versioned"
+)
+
+type IPAMClaimsManager struct {
+	networkClient    networkclient.Interface
+	ipamClaimsClient ipamclaimsclient.Interface
+}
+
+func NewIPAMClaimsManager(networkClient networkclient.Interface, ipamClaimsClient ipamclaimsclient.Interface) *IPAMClaimsManager {
+	return &IPAMClaimsManager{
+		networkClient:    networkClient,
+		ipamClaimsClient: ipamClaimsClient,
+	}
+}
+
+func (m *IPAMClaimsManager) CreateNewPodIPAMClaims(vmi *virtv1.VirtualMachineInstance, ownerRef *v1.OwnerReference) error {
```

Comment:
IMHO we should generalize this function's name. I imagine this will also be called eventually for hotplug / unplug, and the idea is pretty much to reconcile the VMI's spec networks / interfaces w/ the VMI's status interface.

I'm OK with keeping this as is now, and change that once we add support for hotplug/unplug though.

Reply:
Alona asked that now we keep this name, so it will reflect it is used only for new pods (it is under new pod creation condition) and then it can use the new naming scheme